s4e-web/src/app/views/settings/manage-institutions/parent-institution-modal/parent-institution-modal.component.ts

```diff
@@ -0,0 +1,72 @@
+import { InstitutionService } from './../../state/institution.service';
+import { untilDestroyed } from 'ngx-take-until-destroy';
+import { InstitutionQuery } from './../../state/institution.query';
+import { ModalComponent } from './../../../../modal/utils/modal/modal.component';
+import { Component, Inject, OnDestroy, OnInit } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { ModalService } from 'src/app/modal/state/modal.service';
+import { MODAL_DEF } from 'src/app/modal/modal.providers';
+import { assertModalType } from 'src/app/modal/utils/modal/misc';
+import {
+  isParentInstitutionModal,
+  ParentInstitutionModal,
+  PARENT_INSTITUTION_MODAL_ID
+} from './parent-institution-modal.model';
+import { Institution } from '../../state/institution.model';
+import { FormControl } from '@ng-stack/forms';
+
+@Component({
+  selector: 's4e-parent-institution-modal',
+  templateUrl: './parent-institution-modal.component.html',
+  styleUrls: ['./parent-institution-modal.component.scss']
+})
+export class ParentInstitutionModalComponent extends ModalComponent<Institution> implements OnInit, OnDestroy {
+  public institutions: Institution[] = [];
+  public searchedInstitutions: Institution[] = [];
+  public selectedInstitution: Institution = null;
```

Comment:
Generally if we accept that variable can be null then it's better to use type like this: `Institution|null`